### PR TITLE
add announcement for dropping MacOS<10.13

### PR DESCRIPTION
**Pull request by h-vetinari**
First step according to the TODO List [here](https://github.com/conda-forge/conda-forge.github.io/issues/1844#issuecomment-1675953297).

While the stance of upstream CPython does not affect our infrastructure & metadata per se, it's worth noting that [discussion](https://discuss.python.org/t/moving-packaging-and-installers-to-macos-10-13-as-a-minimum/31907) on DPO started by @jakirkham to this effect is still fresh, and it sounds like CPython will also upgrade "in the not too distant future", and might jump even further (e.g. 10.15 or 11.0). We can still choose to follow suit with such a step later on, but moving to 10.13 now is (effectively) necessary to unblock abseil, grpc, protobuf, (and soon) libcxx etc., so IMO we shouldn't wait for however long CPython takes to come to a decision on that.

I've added a second commit that clarifies our support stance (pointing out how it is "best effort"), but I can drop that if anyone disagrees.

### CODE REVIEW

File: src/user/announcements.rst
Comment (h-vetinari) [inline src/user/announcements.rst:12]: TODO: set date before merging.

File: src/user/announcements.rst
Comment (pkgw) [inline src/user/announcements.rst:20]: Is there a specific policy change to mention here? "Starting on <DATE>, the default target MacOS version for new and updated conda-forge packages will be 10.13, and as time passes, it can be expected that conda-forge packages will lose compatibility with older MacOS versions."

In particular, right now the body of the announcement doesn't specifically mention 10.13 as the new minimum — that only appears in the title.

File: src/user/announcements.rst
Comment (pkgw) [inline src/user/announcements.rst:23]: I think this is fine to mention, but the legalistic language might be a bit off-putting for a community project. We could write something like: "... and unfortunately there comes a point when it is no longer practical to support certain systems."

**Comment by beckermr** [comment]
I agree the tone of the note is a bit negative / off putting at times.

**Comment by h-vetinari** [comment]
> I agree the tone of the note is a bit negative / off putting at times.

Happy to reformulate however people wish to

**Comment by h-vetinari** [comment]
Any concrete suggestions how to improve the text @beckermr?

@pkgw, hope your question has been addressed? As I said this is not setting a policy, but we can discuss whether there's appetite for that in the core call.

**Comment by beckermr** [comment]
I'll send some changes. Sorry for the comment and then delay in taking any action.

**Comment by pkgw** [comment]
@h-vetinari My only further suggestion is that the body text should explicitly mention the new minimum version, to go along with what's in the title.

**Comment by hmaarrfk** [comment]
would it be possible to write down the years in which these key versions were released?

**Comment by beckermr** [comment]
Very sorry. Will send comments tonight. Back to school is tomorrow for my kids. It's a bit crazy. If you absolutely need to go ahead, feel free to.

File: src/user/announcements.rst
Comment (beckermr) [inline src/user/announcements.rst:29]:
```suggestion
2023-08-24: Bumping Minimum MacOS version to 10.13
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

We will bump the minimum MacOS version from 10.9 (released in
Oct. 2013, end-of-life since Dec. 2016) to 10.13 (released Sept. 2017, end-of-life since Dec. 2020). Several core packages in the ecosystem either now, or will soon, require at least 10.13. These packages include the C++ standard library for OSX itself, ``libcxx``, starting with version 17. (**TODO: link to RFC from llvm**) This change will not affect already published artifacts, but in the near future, all new builds for OSX will require at least 10.13. This constraint will be implemented through the ``__osx`` virtual package, but the details of how we will achieve this are still being worked out.
```

Comment (beckermr) [reply src/user/announcements.rst:29]: Here is my suggested text @h-vetinari. I went for a brief note explaining the bare minimum and what users can expect.

File: src/user/announcements.rst
Comment (h-vetinari) [inline src/user/announcements.rst:17]: I've committed your suggestion @beckermr, and I'm fine if we leave it at that. I do/did find it relevant to note the relation between the SDK and how we ship libcxx, because I'm pretty sure that this aspect is understood only by very few people; I suspect that most users have no idea we support such old MacOS, or consider it black magic (regulars in conda-forge will have seen `_LIBCPP_DISABLE_AVAILABILITY` but that doesn't mean they understand how this works; at least I didn't for a long time).

Something like the following
```suggestion
The main reason we managed to support 10.9 this long at all, is that conda-forge is able to ship an
up-to-date C++ standard library for OSX, ``libcxx``, in its environments, replacing the old one
in the SDK.

However, several core packages in the ecosystem either now, or will soon, require at least 10.13,
in a way that we cannot be circumvent. These packages include ``libcxx``,
```

Comment (beckermr) [reply src/user/announcements.rst:17]: Sure go for it!

File: src/user/announcements.rst
Comment (h-vetinari) [reply src/user/announcements.rst:25]: Fair point. Though this has been in use for quite a while as _the_ [mechanism](https://conda-forge.org/docs/maintainer/knowledge_base.html#requiring-newer-macos-sdks) for reflecting bumps in `MACOSX_DEPLOYMENT_TARGET`. In other words, users on old MacOS + conda<4.8 will already wrongly get a number of packages that are broken for them.

Conda 4.8 is also [almost](https://github.com/conda/conda/releases/tag/4.8.0) 4 years old, so I doubt that this has a blast radius (it's a subset of MacOS <10.13 users, which is apparently a microscopic minority if the [download numbers](https://discuss.python.org/t/moving-packaging-and-installers-to-macos-10-13-as-a-minimum/31907/5?u=h-vetinari) from pillow are any guide).

Comment (beckermr) [reply src/user/announcements.rst:25]: Yes this is why I mentioned the package in the announcement. If we put it in constrains we can avoid a lot of breakage but we have to decide.

**Review by hmaarrfk — APPROVED** [review]

File: src/user/announcements.rst
Comment (beckermr) [inline src/user/announcements.rst:26]:
```suggestion
virtual package, but the details of how we will achieve this are still being worked out. Only ``conda`` versions 4.8.0 or newer have this virtual package. If you are using a system with MacOS older than 10.13 and are using ``conda`` older than 4.8.0, you will need to either upgrade ``conda`` to at least 4.8.0 or upgrade your system to at least MacOS 10.13.
```

**Review by beckermr — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Comment by dell-datascience** [comment]
in my quest to install apache airflow, i encountered this error which led me here. I hear it has something to do with MACOS-10.9. any headway yet?

this is the error:

Building wheels for collected packages: google-re2, pendulum
  Building wheel for google-re2 (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [15 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-10.9-x86_64-cpython-311
      copying re2.py -> build/lib.macosx-10.9-x86_64-cpython-311
      running build_ext
      building '_re2' extension
      creating build/temp.macosx-10.9-x86_64-cpython-311
      clang -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/air/anaconda3/envs/super/include -fPIC -O2 -isystem /Users/air/anaconda3/envs/super/include -I/usr/local/opt/readline/include -I/Users/air/anaconda3/envs/super/include/python3.11 -c _re2.cc -o build/temp.macosx-10.9-x86_64-cpython-311/_re2.o -fvisibility=hidden
      _re2.cc:11:10: fatal error: 'pybind11/pybind11.h' file not found
      #include <pybind11/pybind11.h>
               ^~~~~~~~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for google-re2
  Running setup.py clean for google-re2
  Building wheel for pendulum (pyproject.toml) ... done
  Created wheel for pendulum: filename=pendulum-2.1.2-cp311-cp311-macosx_11_0_x86_64.whl size=124681 sha256=9d9ae2fbf4746efb4fc52ea8cfabad1891b7b5971cdbaf5cea6c0e59a9c3e8a0
  Stored in directory: /Users/air/Library/Caches/pip/wheels/a4/b7/10/a86f4f49b1bdf8a0cf3da42d48f499e4cb7529ebab7aad3550
Successfully built pendulum
Failed to build google-re2
ERROR: Could not build wheels for google-re2, which is required to install pyproject.toml-based projects

**Pull request merged**
